### PR TITLE
\RestfulFormatterHalJson::moveReferencesToEmbeds only works on multiple value fields

### PR DESCRIPTION
**Pull request by e0ipso**
\RestfulFormatterHalJson::moveReferencesToEmbeds does not work with a field that is configured as being single-value (e.g. taxonomy term reference). (Possibly related: #282)

### CODE REVIEW

**Comment by e0ipso** [comment]
@amitaibu ready to review.

**Comment by amitaibu** [comment]
I hate to be that guy, but maybe worth a test? ;)

**Comment by e0ipso** [comment]
It is worth a test indeed. I was just lazy.

I wouldn't like you if you were not that guy ;-)

**Comment by e0ipso** [comment]
@amitaibu added tests are green now.

File: plugins/formatter/hal_json/RestfulFormatterHalJson.class.php [hunk 1]
```diff
@@ -228,40 +228,59 @@ protected function getCurie() {
...
     // Remove the original reference.
     unset($row[$public_field_name]);
   }
 
-}
+  /**
+   * Moves a single instance to the embeds.
+   *
+   * @param array $output
+   *   Output array to be modified.
```
Comment (amitaibu) [inline plugins/formatter/hal_json/RestfulFormatterHalJson.class.php]: `Output array to be modified. Passed by reference.`

File: plugins/formatter/hal_json/RestfulFormatterHalJson.class.php [hunk 2]
```diff
@@ -228,40 +228,59 @@ protected function getCurie() {
...
+      $this->moveMetadataResource($output, $public_field, $value_metadata, $row[$public_field_name]);
     }
 
     // Remove the original reference.
     unset($row[$public_field_name]);
   }
 
-}
+  /**
+   * Moves a single instance to the embeds.
```
Comment (amitaibu) [inline plugins/formatter/hal_json/RestfulFormatterHalJson.class.php]: Maybe: `Move a single "embedded resource" to be under the "_embedded" property` ?

File: tests/RestfulViewEntityTestCase.test [hunk 3]
```diff
@@ -376,4 +376,40 @@ class RestfulViewEntityTestCase extends RestfulCurlBaseTestCase {
     $this->assertEqual(count($decoded_json['data']), 1, 'The correct number of items was listed.');
     $this->assertNotNull($decoded_json['next'], '"Next" property added.');
   }
+
+  /**
+   * Test embedded resources.
+   */
+  function testHalEmbeddedResources() {
```
Comment (amitaibu) [inline tests/RestfulViewEntityTestCase.test]: `RestfulViewEntityTestCase` is becoming a beast. Maybe we can move this to own class - `RestfulHalJsonTestCase`?

**Comment by amitaibu** [comment]
Thanks. I've added minor comments, after that it can be merged.

**Comment by e0ipso** [comment]
Tests are green. Merging.

File: tests/RestfulSimpleJsonTestCase.test [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * @file
+ * Contains RestfulSimpleJsonTestCase
+ */
+
+class RestfulSimpleJsonTestCase extends RestfulCurlBaseTestCase {
+
+  public static function getInfo() {
+    return array(
+      'name' => 'View HAL+JSON',
+      'description' => 'Test the viewing of an entity in HAL+JSON format.',
```
Comment (drewish) [inline tests/RestfulSimpleJsonTestCase.test:13]: I think the name and description was copy pasta from the other test case.

**Pull request closed without merging**